renderer/reducers/payment.js

```diff
@@ -261,15 +262,39 @@ const ACTION_HANDLERS = {
   },
 }
 
+const paymentSelectors = {}
 const modalPaymentSelector = state => state.payment.payment
+const paymentsSelector = state => state.payment.payments
+const paymentsSendingSelector = state => state.payment.paymentsSending
+const nodesSelector = state => networkSelectors.nodes(state)
 
-const paymentSelectors = {}
+paymentSelectors.payments = createSelector(
+  paymentsSelector,
+  payments => payments
+)
+
+paymentSelectors.paymentsSending = createSelector(
+  paymentsSendingSelector,
+  paymentsSending => paymentsSending
+)
 
 paymentSelectors.paymentModalOpen = createSelector(
   modalPaymentSelector,
   payment => !!payment
 )
 
+paymentSelectors.decoratedPaymentsSelector = createSelector(
+  paymentsSelector,
+  nodesSelector,
+  (payments, nodes) =>
+    payments.map(payment => {
+      const destPubKey = payment.path && payment.path[payment.path.length - 1]
+      payment.dest_node_pubkey = destPubKey ? destPubKey : null
+      payment.dest_node_alias = destPubKey ? getNodeAlias(destPubKey, nodes) : null
+      return payment
```

Comment:
I'd keep this immutable just in case (clone payment and return modified clone)

Reply:
yeah, done